app/models/concerns/smooch_search.rb

```diff
@@ -9,19 +9,20 @@ module ClassMethods
     def search(app_id, uid, language, message, team_id, workflow, provider = nil)
       platform = self.get_platform_from_message(message)
       begin
+        limit = CheckConfig.get('most_relevant_team_limit', 3, :integer)
         sm = CheckStateMachine.new(uid)
         self.get_installation(self.installation_setting_id_keys, app_id) if self.config.blank?
         RequestStore.store[:smooch_bot_provider] = provider unless provider.blank?
         query = self.get_search_query(uid, message)
-        results = self.get_search_results(uid, query, team_id, language).collect{ |pm| Relationship.confirmed_parent(pm) }.uniq
+        results = self.get_search_results(uid, query, team_id, language, limit).collect{ |pm| Relationship.confirmed_parent(pm) }.uniq
```

Comment:
The limit here should be much higher, I think we can even hard code it to 100, or something like that. Because further down we apply the limit over the results that actually have published reports (see comment below).

Reply:
@caiosba the limit work as the existing behavior for example the `get_search_results`  called one of the following methods

- [parse_search_results_from_alegre](https://github.com/meedan/check-api/blob/develop/app/models/concerns/smooch_search.rb#L103-L106) 
- [search_by_keywords_for_similar_published_fact_checks](https://github.com/meedan/check-api/blob/develop/app/models/concerns/smooch_search.rb#L103-L106)

and both methods in the current develop branch limit the results to 3 so in this refactor I passed the limit through the parent method `get_search_results`

so you suggest to send the limit to `get_search_results` with higher value (100 for example) and after apply all filters limit the result to 3 (the limit from config value) right?

my suggestion to keep the current implementation as this refactor run with old one and just read the limit from the config value and file a ticket with a refactor to make sure to return a limit after apply all filter as there is no issue reported that there is an expected result should return by tipline search for example.

Reply:
Understood, @melsawy ... I reviewed the changes since my last review, let me do a full PR re-review in order to be sure that everything is consistent.